geometry/src/aabb.rs

```diff
@@ -29,25 +29,52 @@ impl Aabb {
     /// ```
     /// use myelin_geometry::Aabb;
     ///
-    /// let area = Aabb::new((10.0, 10.0), (20.0, 0.0));
+    /// let area = Aabb::try_new((10.0, 10.0), (20.0, 0.0)).expect("Invalid aabb");
     /// ```
     ///
     /// ## From points
     /// ```
     /// use myelin_geometry::{Aabb, Point};
     ///
-    /// let area = Aabb::new(Point { x: 0.0, y: 10.0 }, Point { x: 20.0, y: 20.0 });
+    /// let area =
+    ///     Aabb::try_new(Point { x: 0.0, y: 10.0 }, Point { x: 20.0, y: 20.0 }).expect("Invalid aabb");
     /// ```
     ///
+    /// # Errors
+    ///
+    /// Returns an error when both points are the same.
+    ///
     /// [`Aabb`]: ./struct.Aabb.html
-    pub fn new<P1, P2>(upper_left: P1, lower_right: P2) -> Self
+    pub fn try_new<P1, P2>(upper_left: P1, lower_right: P2) -> Result<Self, ()>
     where
         P1: Into<Point>,
         P2: Into<Point>,
     {
-        Self {
-            upper_left: upper_left.into(),
-            lower_right: lower_right.into(),
+        let upper_left = upper_left.into();
+        let lower_right = lower_right.into();
+
+        if upper_left >= lower_right {
+            Err(())
+        } else {
+            Ok(Self {
+                upper_left,
+                lower_right,
+            })
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Missing tests: Only upper_left.x is too hight & only upper_left.y is too high

Reply:
Adressed in e4130ce